Terminal.Gui/Views/DateField.cs

```diff
@@ -283,10 +305,52 @@ static int GetFormatIndex (string [] fm, string t)
 		return idx;
 	}
 
+	private string GetDataSeparator (string separator)
+	{
+		var sepChar = separator.Trim ();
+		if (sepChar.Length > 1 && sepChar.Contains ('\u200f')) {
+			sepChar = sepChar.Replace ("\u200f", "");
+		}
+
+		return sepChar;
+	}
+
+	private static string StandardizeDateFormat (string format) =>
```

Comment:
This feels very hacky and fragile. There has got to be a more straight forward and readable way of accomplishing what you're doing here. 

Almost all of these map to the same thing, but some appear to map a `?` to the same char that proceeds the `?`. 

This code is obtuse. If this is really the simplest way of doing this then please include comments explaining the logic.

Reply:
Yeah, I agree that this code is obtuse. However, other, more sophisticated solutions were much more complex, and none of them worked as we needed to handle numerous edge cases. With the current approach, this method gets the job done, and it truly is the simplest way to make it work. However, I admit it's not the most elegant solution.

Reply:
Fine. At least document the heck out of it with comments please.

Reply:
@tig I have already included a comment describing the StandardizeDateFormat method. Is there anything else you would like documented?